src/Nethermind/Nethermind.Network/P2P/Subprotocols/NodeData/Messages/NodeDataMessageSerializer.cs

```diff
@@ -0,0 +1,41 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using DotNetty.Buffers;
+using Nethermind.Serialization.Rlp;
+
+namespace Nethermind.Network.P2P.Subprotocols.NodeData.Messages;
+
+public class NodeDataMessageSerializer : IZeroInnerMessageSerializer<NodeDataMessage>
+{
+    public void Serialize(IByteBuffer byteBuffer, NodeDataMessage message)
+    {
+        int length = GetLength(message, out int contentLength);
+        byteBuffer.EnsureWritable(length, true);
+        RlpStream rlpStream = new NettyRlpStream(byteBuffer);
```

Comment:
Should we dispose `NettyRlpStream`? Probably not as it would release the `byteBuffer`.